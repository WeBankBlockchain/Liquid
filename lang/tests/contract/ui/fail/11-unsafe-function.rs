use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub unsafe fn noop(&self) {}
    }
}

fn main() {}
