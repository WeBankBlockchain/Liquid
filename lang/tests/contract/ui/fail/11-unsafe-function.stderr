error: `unsafe` is not supported for methods in contract
  --> $DIR/11-unsafe-function.rs:12:13
   |
12 |         pub unsafe fn noop(&self) {}
   |             ^^^^^^
